### ISSUE TITLE
Crash on Debian 9: AttributeError: 'PosixPath' object has no attribute 'seek'

### ISSUE THREAD

**Issue by ecm-pushbx** [dosemu2/install-freedos#7]
I did the following:

* cloned this repo

* `pip3 install tqdm`

* in install-freedos directory: `DESTDIR=$HOME/local.new prefix=/. make install`

* start `DOSEMU2_EXTRAS_DIR=~/local.new/share/dosemu2-extras/ PYTHONPATH=~/local/lib/python3.9/site-packages/ ~/local.new/bin/dosemu -dumb`

* change to drive `e:`

* `insfdusr`

The output ends with this exception:

```
Downloading https://www.ibiblio.org/pub/micro/pc-stuff/freedos/files/distributions/1.2/repos/base/swsubst.zip...
100% 124022/124022 [00:00<00:00, 411798.54B/s]
Downloading https://www.ibiblio.org/pub/micro/pc-stuff/freedos/files/distributions/1.2/repos/base/tree.zip...
100% 75827/75827 [00:00<00:00, 371600.66B/s]
Traceback (most recent call last):
  File "[...]/local.new/libexec/dosemu/dosemu-setupfreedos", line 135, in <module>
    for entry in zipfile.ZipFile(filename).namelist():
  File "/usr/lib/python3.5/zipfile.py", line 1026, in __init__
    self._RealGetContents()
  File "/usr/lib/python3.5/zipfile.py", line 1090, in _RealGetContents
    endrec = _EndRecData(fp)
  File "/usr/lib/python3.5/zipfile.py", line 241, in _EndRecData
    fpin.seek(0, 2)
AttributeError: 'PosixPath' object has no attribute 'seek'
E:\>
```

**Comment by ecm-pushbx** [comment]
I see that I specified python 3.9 for my pip module path but the built-in zipfile module seems to be python 3.5. Is this possibly the cause of the error?

**Comment by stsp** [comment]
I misguided you when said
that there is no such problem
with freedos installed with the
script.
Sorry.
The problem is that you can't
put your hdimage to drive J:
or alike.
Putting your hdimage before
automatic drives, will affect
freedos install in either case.

**Comment by stsp** [comment]
Or actually maybe I didn't mis-gude you,
see https://github.com/dosemu2/dosemu2/discussions/1456#discussioncomment-600277

**Comment by ecm-pushbx** [comment]
I managed to avoid this error by running the following:

`perl -i -pe 's/#!\s*\/usr\/bin\/python3/#![...]\/local\/bin\/python3.9/' ~/local.new/libexec/dosemu/*`

So instead of the python3 that's system wide it now uses my own python3.9. This also means I no longer have to set PYTHONPATH.

**Comment by stsp** [comment]
If this is a problem with newer
python, please re-open.

**Comment by jschwartzenberg** [comment]
I attempted to make sure all scripts were compatible with the Python 3 version that's included with Ubuntu 18.04. I presumed that all higher versions would be backwards compatible, so far I did not run into issues. 

Sometimes rolling distributions have difficulty with Python updates where some modules are still in directories relate to older Python versions and the newer version is not searching there for modules.

Note that `python3-tqdm` should be a package on Debian. I would not recommend using `pip` to install any Python modules. These scripts only rely on modules that are packaged by all major distributions.
